grails-app/services/au/org/ala/merit/AbnLookupService.groovy

```diff
@@ -53,4 +59,78 @@ class AbnLookupService {
         }
         return result
     }
+
+    /**
+     * Validates the ABN format and will do a calculation based on the ABN validation algorithm https://abr.business.gov.au/Help/AbnFormat
+     *
+     * @param abnString
+     * @return
+     */
+    private static Boolean validateABN(String abnString) {
+        abnString = getNonBlankNumericStringWithoutWhitespace(abnString);
+        if(abnString == null) {
+            return false;
+        }
+
+        if(abnString.length() != 11) {
+            return false;
+        }
+
+        if(abnString.substring(0,1) == '0') {
+            return false;
+        }
+
+        String subtract1 = String.valueOf(Long.valueOf(abnString.substring(0,1))-1);
+        String modifiedABN = String.valueOf(subtract1 + abnString.substring(1));
+        Long abnWeightingSum = calcWeightingSum(modifiedABN);
+        Long modEightyNineRemainder = Math.floorMod(abnWeightingSum, 89);
+
+        if(modEightyNineRemainder != 0) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * The 11 digit ABN is structured as a 9 digit identifier with two leading check digits.
+     * The leading check digits are derived using a modulus 89 (remainder after dividing by 89) calculation.
+     *
+     * @param theNumString
+     * @return
+     */
+    private static Long calcWeightingSum(String theNumString) {
+        List<Integer> weightList =  [10, 1, 3, 5, 7, 9, 11, 13, 15, 17, 19];
+        Long weightingSum = 0;
+
+        Integer startIndex = 0;
+        Integer endIndex = 10;
+
+        for(Integer i = startIndex; i <= endIndex; i++) {
+            weightingSum +=
+                    ( Long.valueOf(theNumString.substring(i,i+1) ) * weightList[i]);
+        }
+
+        return weightingSum;
+    }
+
+    /**
+     * Validates if the parameter is blank and will remove whitespaces
+     *
+     * @param theString
+     * @return
+     */
+    private static String getNonBlankNumericStringWithoutWhitespace(String theString) {
+        if(StringUtils.isBlank(theString)) {
+            return null;
+        }
+
+        theString = StringUtils.deleteWhitespace(theString)
```

Comment:
I am not sure if the ABN service will delete whitespace or not.  If we are going to do this (which makes sense to do) we should use the "de-whitespaced" string that passed validation when calling the service.

Reply:
That sounds good, I'll pass the "de-whitespaced" string into the service instead of just returning boolean.